### PR TITLE
[samurailegends-generals-balance] add generals balance test for samurai legends

### PR DESCRIPTION
**Pull request by ThijmenMaus**
This commits adds the Samurai Legends Generals voting strategy. The voting power is determined by how many generals a user has. Calculated from a batch nft contract.

### CODE REVIEW

File: src/strategies/samurailegends-generals-balance/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { BigNumberish } from '@ethersproject/bignumber';
+import { Multicaller } from '../../utils';
+
+export const author = 'Samurai-Legends';
+export const version = '0.1.2';
+
+const abi = [
+  'function erc721GetAllTokensOfOwner(address nftAddress, address user) external view returns (uint[] memory)'
+];
```
Comment (ChaituVR) [inline src/strategies/samurailegends-generals-balance/index.ts:8]: Hey @ThijmenMaus Why not use https://snapshot.org/#/strategy/erc721 strategy? 🤔

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/samurailegends-generals-balance/index.ts (hunk 1 above)
Comment (ThijmenMaus) [reply src/strategies/samurailegends-generals-balance/index.ts:8]: Is it possible to apply that together with the limitations of max id 5000?

Comment (ThijmenMaus) [reply src/strategies/samurailegends-generals-balance/index.ts:8]: @ChaituVR

Comment (ChaituVR) [reply src/strategies/samurailegends-generals-balance/index.ts:8]: Oh i see it now, i read the Readme and thought this is just getting balance of users

**Review by ChaituVR — APPROVED** [review]

File: src/strategies/samurailegends-generals-balance/index.ts (hunk 1 above)
Comment (ThijmenMaus) [reply src/strategies/samurailegends-generals-balance/index.ts:8]: Thanks for the information, I will remember to add information regarding that in the readme next time.

**Pull request merged**